x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java

```diff
@@ -545,6 +546,7 @@ public List<NamedWriteableRegistry.Entry> getNamedWriteables() {
             ),
             // Data Streams
             new NamedWriteableRegistry.Entry(XPackFeatureSet.Usage.class, XPackField.DATA_STREAMS, DataStreamFeatureSetUsage::new),
+            new NamedWriteableRegistry.Entry(XPackFeatureSet.Usage.class, XPackField.DATA_LIFECYCLE, DataLifecycleFeatureSetUsage::new),
```

Comment:
I wasn't sure -- I always loaded it because the API has an `enabled` field. But I guess that enabled field is not actually meant for feature flags, is it? I'll hide all of these if the feature flag is off.